asset/mock.go

```diff
@@ -159,7 +159,10 @@ func virtualGenesisTx(newAsset *Asset) (*wire.MsgTx, error) {
 	// With our single input and output mapped, we're ready to construct our
 	// virtual transaction.
 	virtualTx := wire.NewMsgTx(2)
-	virtualTx.AddTxIn(txIn)
+
```

Comment:
So this can't be done unconditionally. We need to make a new TAP VM version, then flatten the virtual transaction _only_ for that version.